Essentials/src/main/java/com/earth2me/essentials/EssentialsUpdateChecker.java

```diff
@@ -0,0 +1,258 @@
+package com.earth2me.essentials;
+
+import com.google.common.base.Charsets;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonSyntaxException;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+import static com.earth2me.essentials.I18n.tl;
+
+public final class EssentialsUpdateChecker {
```

Comment:
Possibly move to `com.earth2me.essentials.updatecheck.UpdateChecker`? Base package is cluttered as it is.